bot/code_review_bot/__init__.py

```diff
@@ -167,33 +167,39 @@ def hash(self):
         """
         assert self.revision is not None, "Missing revision"
 
-        local_repository = settings.runtime.get("mercurial_repository")
-        if local_repository:
-            logger.debug("Using the local repository to build issue's hash")
-            try:
-                with open(local_repository / self.path) as f:
-                    file_content = f.read()
-            except (FileNotFoundError, IsADirectoryError):
-                logger.warning("Failed to find issue's related file", path=self.path)
-                file_content = None
-        else:
-            try:
-                # Load all the lines affected by the issue
-                file_content = self.revision.load_file(self.path)
-            except ValueError:
-                # Build the hash with an empty content in case the path is erroneous
-                file_content = None
-            except requests.exceptions.HTTPError as e:
-                if e.response.status_code == 404:
+        # Build the hash only if the file is not autogenerated.
+        # An autogenerated file resides in the build directory that it has the
+        #  format `obj-x86_64-pc-linux-gnu`
+        file_content = None
+        if "/obj-x86_64-pc-linux-gnu/" not in self.path:
+            local_repository = settings.runtime.get("mercurial_repository")
+            if local_repository:
+                logger.debug("Using the local repository to build issue's hash")
+                try:
+                    with open(local_repository / self.path) as f:
+                        file_content = f.read()
+                except (FileNotFoundError, IsADirectoryError):
                     logger.warning(
-                        "Failed to download a file with an issue", path=self.path
+                        "Failed to find issue's related file", path=self.path
```

Comment:
After logging this warning, we still need to set "file_content = None" as before.